### PR TITLE
feat: create @looker/extension-utils

### PR DESCRIPTION
**Pull request by jkaster**
Moved the "extension provider" browser compatibility interface and functions from API Explorer into a new package called `@looker/extension-utils`

- Renamed things to make them more generic
- Applied `@looker/extension-utils` to:
  - packages/api-explorer
  - packages/extension-api-explorer
  - packages/hackathon (which now has "standard" branding behavior)
- documented README.md
- updated various CI workflows

## Hackathon style update

image

### CODE REVIEW

File: packages/extension-utils/src/browserAdaptor.ts [hunk 1]
```diff
@@ -24,23 +24,41 @@
...
-    </Provider>
-  )
-})
+import type { IExtensionAdaptor, ThemeOverrides } from './adaptorUtils'
+import { getThemeOverrides, hostedInternally } from './adaptorUtils'
+
+/**
+ * An adaptor class for interacting with browser APIs when not running in an extension
+ */
+export class BrowserAdaptor implements IExtensionAdaptor {
```
Comment (jkaster) [inline packages/extension-utils/src/browserAdaptor.ts:33]: Ignore that git thinks this was a migration from `App.tsx`. That's completely and utterly bogus.

File: packages/extension-utils/src/index.ts [hunk 2]
```diff
@@ -23,9 +23,7 @@
...
  */
-import type { IApixEnvAdaptor } from '../utils'
-import { registerEnvAdaptor, StandaloneEnvAdaptor } from '../utils'
 
-export const registerTestEnvAdaptor = (envAdaptor?: IApixEnvAdaptor) => {
-  registerEnvAdaptor(envAdaptor || new StandaloneEnvAdaptor())
-}
+export * from './adaptorUtils'
+export * from './browserAdaptor'
+export * from './extensionAdaptor'
```
Comment (jkaster) [inline packages/extension-utils/src/index.ts:29]: Wow, another bogus git "migration"

File: packages/run-it/src/utils/RunItSDK.ts [hunk 3]
```diff
@@ -137,3 +137,7 @@ export const sdkNeedsAuth = (sdk: IAPIMethods | undefined) => {
   const configIsNeeded = sdkNeedsConfig(sdk)
   return configIsNeeded && !sdk.authSession.isAuthenticated()
 }
+
+export enum StoreConstants {
+  LOCALSTORAGE_SETTINGS_KEY = 'settings',
+}
```
Comment (jkaster) [inline packages/run-it/src/utils/RunItSDK.ts:143]: Moved this out of extension provider into RunIt which has a "base" redux store that can be used with API Explorer also

**Comment by drstrangelooker** [comment]
LGTM

**Review by drstrangelooker — APPROVED** [review]

File: packages/extension-utils/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+yarn add @looker/extension-utils
+```
+
+or
+
+```shell
+npm install @looker/extension-utils
+```
+
+## Using extension adaptors
```
Comment (bryans99) [inline packages/extension-utils/README.md:23]: I think this should be envirobment adapters

File: packages/extension-utils/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+All source code for the application except for the launch page can be the same. For the launch page, either `ExtensionAdaptor` or `BrowserAdaptor` will be used.
+
+### BrowserAdaptor
+
+See [StandAloneApiExplorer.tsx](https://github.com/looker-open-source/sdk-codegen/tree/main/packages/api-explorer/src/StandAloneApiExplorer.tsx) for a reference implementation.
+
+### ExtensionAdaptor
```
Comment (bryans99) [inline packages/extension-utils/README.md:31]: ExtensionAdapter is the CORRECT name for the concrete implementation. The interface should be IEnvironmentAdaptor

File: packages/extension-utils/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  "scripts": {
+    "analyze": "export ANALYZE_MODE=static && yarn bundle",
+    "bundle": "tsc && webpack --config webpack.prod.config.js",
+    "deploy": "bin/deploy",
+    "develop": "webpack serve --hot --disable-host-check --port 8080 --https --config webpack.dev.config.js",
+    "watch": "yarn lerna exec --scope @looker/extension-utils --stream 'BABEL_ENV=build babel src --root-mode upward --out-dir lib/esm --source-maps --extensions .ts,.tsx --no-comments --watch'"
+  },
+  "dependencies": {
+    "@looker/components": "^2.8.1",
+    "@looker/extension-sdk": "^21.18.1",
```
Comment (bryans99) [inline packages/extension-utils/package.json:24]: Should these versions be "*"?

**Review by bryans99 — APPROVED** [review]
Approving but consider renaming IExtensionAdaptor to IEnvironmentAdaptor

File: packages/extension-utils/package.json (hunk 6 above)
Comment (jkaster) [reply packages/extension-utils/package.json:24]: I think it gets complicated when it's *also* a package that gets published out of the monorepo? But you'd probably know better than me. Certainly worth testing in a future PR.

File: packages/extension-utils/README.md (hunk 4 above)
Comment (jkaster) [reply packages/extension-utils/README.md:23]: Ok. I'll do the rename refactor and put it back. Makes sense.

**Review by bryans99 — DISMISSED** [review]
You missed one reference in the README

**Comment by jkaster** [comment]
> You missed one reference in the README

@bryans99 I'm not seeing the miss in the README. Can you be more specific?

**Review by bryans99 — APPROVED** [review]
LGTM

**Pull request merged**